app/lib/fake/backend/fake_pana_runner.dart

```diff
@@ -112,3 +114,26 @@ class FakePanaRunner implements PanaRunner {
     );
   }
 }
+
+class _HashGenerator {
+  final String package;
+  final String version;
+  final _keys = <String>{};
+
+  _HashGenerator(this.package, this.version);
+
+  /// Returns the hash of the [key]. When [max] is present, only
+  /// ints between 0 and max (exclusive) will be returned.
+  ///
+  /// Throws [StateError] if it is called more than once with teh same [key].
+  int hashOf(String key, [int? max]) {
```

Comment:
This is a bit of a hack, but works... maybe we can find a better name than `_HashGenerator` but that's all..
Or make functions:
```

int Function(String input, {int? max}) createHasher(String seed) => (input, {int? max}) {
   ....
}
```